src/core/utils/spec-preprocess.ts

```diff
@@ -34,12 +37,15 @@ import { getStyleOverridden } from '../utils/style';
  * @param callback
  */
 export function traverseTracks(
-    spec: GoslingSpec,
+    spec: GoslingSpec | View | PartialTrack,
     callback: (t: Partial<Track>, i: number, ts: Partial<Track>[]) => void
 ) {
     if ('tracks' in spec) {
-        spec.tracks.forEach((...args) => callback(...args));
-    } else {
+        spec.tracks.forEach((t, i, ts) => {
```

Comment:
Updates on this file are not directly related to this PR but are to just handle some cases that were not handled correctly (i.e., `tracks` in a track).